src/main/java/org/springframework/cloud/gateway/filter/headers/XForwardedHeadersFilter.java

```diff
@@ -0,0 +1,276 @@
+package org.springframework.cloud.gateway.filter.headers;
+
+import java.net.URI;
+import java.util.LinkedHashSet;
+import java.util.List;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.core.Ordered;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.util.StringUtils;
+import org.springframework.web.server.ServerWebExchange;
+
+import static org.springframework.cloud.gateway.support.ServerWebExchangeUtils.GATEWAY_ORIGINAL_REQUEST_URL_ATTR;
+import static org.springframework.cloud.gateway.support.ServerWebExchangeUtils.GATEWAY_REQUEST_URL_ATTR;
+
+@ConfigurationProperties("spring.cloud.gateway.x-forwarded")
+public class XForwardedHeadersFilter implements HttpHeadersFilter, Ordered {
```

Comment:
For these overridden files can we:
* Restore the header (we can't copy and paste classes from other projects and drop headers)
* Restore the formatting to spaces rather than tabs (which is in alignment with Alfresco's formatting guidelines and makes it easier to diff against the original)
* Add to the Javadoc description with a brief indication on why the class is overridden, pointing to the individual GitHub issue (in addition to the link in the README), and a TODO line to remove the class in the future when fixed upstream
* Create a GitHub issue in this project to remind us to remove this class in the future when the issue when fixed upstream